### ISSUE TITLE
NFC Tag

### ISSUE THREAD

**Issue by schloempe** [renyuneyun/Easer#191]
I create an event for specified NFC tag id. But phone will always show "new tag" and Easer does not react. How can I make Easer react for NFC tag?

**Comment by renyuneyun** [comment]
Hi,
Could you clarify that a bit more (e.g. through screenshots)? I don't quite follow what you say...

One think maybe worth asking: did you added that specific tag to Easer on the page waiting for an NFC tag?

**Comment by schloempe** [comment]
Yes I add the tag with button DETECT TAG. This works and shows tag id. But if I touch the tag later, only NFC service comes up and say NEW TAG. Your app is not react to the same NFC tag.
How can I add screenshots here?

**Comment by schloempe** [comment]
![screenshot_20181102-113608 1](https://user-images.githubusercontent.com/2212457/47950332-04c27d00-df51-11e8-8fdc-8ffad20605dd.png)
![screenshot_20181102-113645 1](https://user-images.githubusercontent.com/2212457/47950344-2cb1e080-df51-11e8-8d16-44913226d530.png)
![screenshot_20181102-113725 1](https://user-images.githubusercontent.com/2212457/47950349-3c312980-df51-11e8-858f-d410b68b4c4c.png)

**Comment by renyuneyun** [comment]
Seems you are configuring correct.
Does your NFC tag has data / payload? #92 is related to this.

**Comment by schloempe** [comment]
Yes, #92 looks similar. But as you can see in my screenshot, I did write text AUDI on this tag. Before I tried an application link. Both is not working well. So I do not understand. I write AUDI in this  tag,  but Android default NEW TAG DETECTED comes up. Why?

**Comment by schloempe** [comment]
Do I need special NDEF to launch easer? Perhaps a predefined MIME?

**Comment by renyuneyun** [comment]
Hi,
Sorry for the long delay.

I don't think you need a special NDEF for the card, at least in theory. But currently, if I understand Android's mechanism and NFC correctly, the NFC Event in Easer works with an NFC tag with an unknown category to Android (e.g. a bank card, a staff/student card, which was not supposed to be read by Android).
(Please do correct me if I'm possibly wrong. This may help me figure out the exact problem.)

IIRC, I think this is due to Android's way of dispatching NFC tag information to apps, and Easer didn't handle this well (i.e. some scanned tags are not catched, because they belong to other categories). I should investigate this more.
As you may already be aware, I'm not an expert on NFC-related things (and I'm completely confused what, how and why the different "technologies" are there). So this may take longer than expected...

**Comment by ploth** [comment]
Same here, "new tag collected" always pops up. easer doesn't react to the tag.

**Comment by 1e-9** [comment]
Hi.

Same problem with NFC-tags: 
A tag with ISO-14443-4 works. 
These do not work: ISO 14443-3A 
(
I bought this ones: 
https://www.nfc-tag-shop.de/en/nfc-sticker/black-white-nfc-tags/111/nfc-sticker-20-x-15-mm-ntag-213-180-byte-white 
https://www.nfc-tag-shop.de/en/nfc-keyfobs/nfc-keys-pvc/325/nfc-keyfob-30-mm-ntag-213-180-byte-black
)

![works](https://user-images.githubusercontent.com/50927022/58201827-5b698b80-7cd6-11e9-968e-44f9bc37773e.png)

![doesnt_work](https://user-images.githubusercontent.com/50927022/58201871-776d2d00-7cd6-11e9-8206-972bbb4b3b91.png)

**Comment by ildar** [comment]
Play with disabling other apps that handle NFC. Just a wild guess

**Comment by renyuneyun** [comment]
@1e-9 Thanks for providing more technical info!
Unfortunately I don't currently have much time to learn that thing, so the issue is still postponed...

I'll summarise some information here in case anyone is thinking of help:
* The component (skill) for hendling NFC scanning is in package `ryey.easer.skills.event.nfc_tag`
* There is an `Activity` (`NfcListenerActivity`) which handles the NFC scan event (dispatched by Android). The types of NFC tag *technologies* it handles are declared in the corresponding `activity` element in `AndroidManifest.xml`
* Android classifies different types of NFC tags into different *technologies*, and handles them in a separate flow (refer to Android dev documentation for NFC)
* 
* My personal feeling is that this `Activity` should handle more (all) technologies, in the correct way (which I'm not sure)
